hooks/cache_location.py

```diff
@@ -76,36 +76,28 @@ def _get_cache_root(self, project_id, pipeline_configuration_id):
         and inside this base level hook. This method calculates the cache root
         for the current project and configuration. In the default implementation,
         all the different types of cache data resides below a common root point. 
-        
+
         :param project_id: The shotgun id of the project to store caches for
         :param pipeline_configuration_id: The shotgun pipeline config id to store caches for
         :returns: The calculated location for the cache root
         """
-        
         # the default implementation will place things in the following locations:
         # macosx: ~/Library/Caches/Shotgun/SITE_NAME/project_xxx/config_yyy
         # windows: $APPDATA/Shotgun/SITE_NAME/project_xxx/config_yyy
         # linux: ~/.shotgun/SITE_NAME/project_xxx/config_yyy
-        
+
         # first establish the root location
+
         tk = self.parent
-        if sys.platform == "darwin":
-            root = os.path.expanduser("~/Library/Caches/Shotgun")
-        elif sys.platform == "win32":
-            root = os.path.join(os.environ["APPDATA"], "Shotgun")
-        elif sys.platform.startswith("linux"):
-            root = os.path.expanduser("~/.shotgun")
 
-        # get site only; https://www.foo.com:8080 -> www.foo.com
-        base_url = urlparse.urlparse(tk.shotgun.base_url)[1].split(":")[0]
-        
+        root = tk.get_site_cache_root()
```

Comment:
I think local_site_cache_location and being a property are much better actually, good points.

I put it there because the method needs some context to operate with, namely the site you are using. Since the site information is on the Tank object, I thought of adding it there. It's more like a convenience function. You already have your context (the project&site) and you want to know the site location for it. That method is then available for you to call.

If you feel like it muddies the API, I'd be ok just leaving it at the tank.api level and have the caller pass the base url.